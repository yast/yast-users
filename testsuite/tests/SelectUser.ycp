{
    // testedfiles: Users.pm

    import "Users";
    import "UsersCache";
    import "Mode";
    import "Directory";
    import "Progress";
    import "Report";

    string tmpdir	= Directory::tmpdir;
    foreach (string file, [ "passwd", "group", "shadow" ], {
	SCR::Execute (.target.bash, sformat ("/bin/cp ./%1 %2/", file, tmpdir));
    });
    Users::SetBaseDirectory (tmpdir);
    Users::ReadLocal ();

    import "Testsuite";

    map READ = $[
	"etc" : $[
	    "fstab": [],
	    "cryptotab": [],
	    "default": $[
		"useradd": $[
		    "home":	"/home",
		    "groups":	"audio,video",
		    "expire":	nil,
		    "group":	100
		]
	    ],
	],
	"target": $[
	    "stat"	: $[],
	    "size"	: 0
	],
	"product": $[
		"features": $[
		    "USE_DESKTOP_SCHEDULER"	: "no",
		    "IO_SCHEDULER"		: "",
		    "ENABLE_AUTOLOGIN"		: "false",
		    "UI_MODE"			: "simple",
		    "EVMS_CONFIG"		: "no",
		    "INCOMPLETE_TRANSLATION_TRESHOLD"	: "99",
		]
	],
	"sysconfig"	: $[
	    "displaymanager"	: $[
		"DISPLAYMANAGER"	: ""
	    ],
	],
    ];
    map WRITE = $[];
    map EXEC = $[
	"passwd" : $[
	    "init"	: true
	],
	"target" : $[
	    "bash"		: -1,
	    "bash_output"	: $[]
	],
    ];

    Testsuite::Dump ("==========================================================");

    Mode::SetTest ("test");

    Testsuite::Test (``(Users::Read ()), [READ, WRITE, EXEC], 0);

    Testsuite::Test(``(Users::SelectUser (500)), [], 0);

    Testsuite::Dump (sformat ("selected user with id 500:\n %1", Users::GetCurrentUser()));

    Testsuite::Dump ("==========================================================");

    Testsuite::Test(``(Users::SelectUser (25)), [], 0);

    Testsuite::Dump (sformat ("selected user with id 25:\n %1",Users::GetCurrentUser ()) );

    Testsuite::Dump ("==========================================================");

    Testsuite::Test(``(Users::SelectUser (5001)), [], 0);

    Testsuite::Dump (sformat("selected user with id 5001:\n %1",Users::GetCurrentUser ()));

}
