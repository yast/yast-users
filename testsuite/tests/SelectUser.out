Dump	==========================================================
Read	.etc.default.useradd."expire" nil
Read	.etc.default.useradd."group" 100
Read	.etc.default.useradd."groups" "audio,video"
Read	.etc.default.useradd."home" "/home"
Read	.etc.default.useradd."inactive" 0
Read	.etc.default.useradd."shell" 0
Read	.etc.default.useradd."skel" 0
Read	.etc.default.useradd."umask" 0
Read	.target.string "/etc/shells" 0
Read	.etc.nsswitch_conf.passwd 0
Read	.etc.nsswitch_conf.passwd 0
Write	.target.ycp "/var/lib/YaST2/users.ycp" $[] true
Return	
Return	nil
Dump	selected user with id 500:
Dump	 $["addit_data":"", "cn":"HaHa", "gidNumber":"100", "grouplist":$[], "groupname":"users", "homeDirectory":"/home/hh", "loginShell":"/bin/bash", "shadowExpire":"", "shadowFlag":"", "shadowInactive":"", "shadowLastChange":"13727", "shadowMax":"99999", "shadowMin":"0", "shadowWarning":"7", "type":"local", "uid":"hh", "uidNumber":"500", "userPassword":"heslo"]
Dump	==========================================================
Return	nil
Dump	selected user with id 25:
Dump	 $["addit_data":"", "cn":"Batch jobs daemon", "gidNumber":"25", "grouplist":$[], "groupname":"", "homeDirectory":"/var/spool/atjobs", "loginShell":"/bin/bash", "shadowExpire":"", "shadowFlag":"", "shadowInactive":"", "shadowLastChange":"13636", "shadowMax":"99999", "shadowMin":"0", "shadowWarning":"7", "type":"system", "uid":"at", "uidNumber":"25", "userPassword":"!"]
Dump	==========================================================
Return	nil
Dump	selected user with id 5001:
Dump	 $[]
